compiler/src/iree/compiler/Codegen/Common/CPU/CPUMaterializeEncodingPass.cpp

```diff
@@ -155,50 +158,42 @@ enumerateMatmulTileX86_64(EncodingUser user, TypeRange elementTypes,
         };
       }
     }
-    // Note: 16-bit floating point types currently use the same tile size as
-    // f32. This makes sense when either (1) the accumulator is f32, or (2)
-    // the arithmetic will have to expand f16 to f32 in registers. We may
-    // reconsider when taking advantage of native f16/bf16 arithmetic when the
-    // accumulator itself is f16/bf16.
-    if (hasFeature(target, "+avx512f")) {
-      if (hasUkernel(target)) {
+    if ((lhs.isBF16() && rhs.isBF16()) || (lhs.isF16() && rhs.isF16()) ||
+        (lhs.isF32() && rhs.isF32())) {
+      // Note: 16-bit floating point types currently use the same tile size as
+      // f32. This makes sense when either (1) the accumulator is f32, or (2)
+      // the arithmetic will have to expand f16 to f32 in registers. We may
+      // reconsider when taking advantage of native f16/bf16 arithmetic when the
+      // accumulator itself is f16/bf16.
+      if (hasFeature(target, "+avx512f")) {
         return {
             TileMxNxK{16, 16, 1}, // Aim to use VFMADD* (zmm).
             TileMxNxK{8, 16, 1},  // Truncation of the above.
             TileMxNxK{4, 16, 1},  // Truncation of the above.
             TileMxNxK{2, 16, 1},  // Truncation of the above.
             TileMxNxK{1, 16, 1},  // Truncation of the above.
         };
-      } else {
-        // Code generation tile sizes.
+      }
+      if (hasFeature(target, "+avx")) {
+        // Note: for good performance, most +avx users will also want to add
+        // +fma, but that's a local instruction selection detail and the tile
+        // layout is unaffected, as there are enough registers even with the
+        // need for intermediate product registers when +fma is not used.
         return {
-            TileMxNxK{16, 16, 1}, // Aim to use VFMADD* (zmm).
-            TileMxNxK{8, 32, 1},  // Use same number of accumulators.
-            TileMxNxK{4, 64, 1},  // Use same number of accumulators.
-            TileMxNxK{2, 64, 1},  // Use half the number of accumulators.
-            TileMxNxK{1, 128, 1}, // Use half the number of accumulators.
```

Comment:
The logic about ukernel and non-ukernel tile size selection is gone, should we add them back?